docker/src/main/java/org/jclouds/docker/compute/options/DockerTemplateOptions.java

```diff
@@ -384,7 +400,7 @@ public static DockerTemplateOptions overrideLoginCredentials(LoginCredentials cr
       }
 
       /**
-       * @see TemplateOptions#blockUntilRunning
+       * @see TemplateOptions#blockUntilRunning(boolean)
        */
       public static DockerTemplateOptions blockUntilRunning(boolean blockUntilRunning) {
          DockerTemplateOptions options = new DockerTemplateOptions();
```

Comment:
A quick look to all existing template options subclasses show that the only ones that follow this pattern are the ones for Docker, SoftLayer and Azure Compute.
Given that this is not the common pattern and that this PR already cleans up the options in Docker, I think it's better to use this PR to remove the optionals from Docker, and fix SoftLayer and Azure in different PRs.